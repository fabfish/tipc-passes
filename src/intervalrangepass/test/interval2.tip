main() {
  var x, y;
  x = 42;	// x = [42,42]
  y = x + 2;	// y = [44,44]
  return x + y;	// [86,86]
}
