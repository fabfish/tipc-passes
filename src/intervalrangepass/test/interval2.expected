*** interval range analysis for function define i64 @_tip_main() {
entry:
  %addtmp = add i64 42, 2
  %addtmp4 = add i64 42, %addtmp
  ret i64 %addtmp4
}
 ***
  %addtmp4 = add i64 42, %addtmp = [86,86]
  %addtmp = add i64 42, 2 = [44,44]
